File: src/main/java/ar/edu/um/haberes/rest/model/LegajoBanco.java
/**
 * 
 */
package ar.edu.um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.JoinColumns;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "legajoId", "anho", "mes", "cbu" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class LegajoBanco extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -6958775655040740839L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long legajoBancoId;

	private Long legajoId;
	private Integer anho = 0;
	private Integer mes = 0;
	private String cbu = "";
	private BigDecimal fijo = BigDecimal.ZERO;
	private BigDecimal porcentaje = BigDecimal.ZERO;
	private Byte resto = 0;
	private BigDecimal acreditado = BigDecimal.ZERO;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumns({
			@JoinColumn(name = "legajoId", referencedColumnName = "legajoId", insertable = false, updatable = false),
			@JoinColumn(name = "anho", referencedColumnName = "anho", insertable = false, updatable = false),
			@JoinColumn(name = "mes", referencedColumnName = "mes", insertable = false, updatable = false), })
	private Liquidacion liquidacion;

}

File: src/main/java/um/haberes/rest/kotlin/model/LegajoBanco.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["legajoId", "anho", "mes", "cbu"])])
data class LegajoBanco(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var legajoBancoId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var cbu: String = "",
    var fijo: BigDecimal = BigDecimal.ZERO,
    var porcentaje: BigDecimal = BigDecimal.ZERO,
    var resto: Byte = 0,
    var acreditado: BigDecimal = BigDecimal.ZERO,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumns(
        JoinColumn(name = "legajoId", referencedColumnName = "legajoId", insertable = false, updatable = false),
        JoinColumn(name = "anho", referencedColumnName = "anho", insertable = false, updatable = false),
        JoinColumn(name = "mes", referencedColumnName = "mes", insertable = false, updatable = false)
    )
    var liquidacion: Liquidacion? = null

) : Auditable()
